calculator.js

```diff
@@ -1,28 +1,65 @@
 
-const exampleAdditionInput = {
-  num1: 3,
-  num2: 5,
-  operation: 'add',
-}
 
-const calculateUserInput = function (error, promptInput) {
-  console.log('This is the value of the promptInput variable that got passed in by prompt, after our prompt package collect user input', promptInput);
-
-  // Questions to ask and answer:
-  // What is promptInput?
-  // What data type? What does it hold? What does it represent?
-  // How do we read values from it? What syntax?
-  // How can we use it?
-  // Can we call our existing functions now, inside of this function?
-}
-
-// Example manual testing of calculator.  
-// calculateUserInput({}, {
-//   num1: 3,
-//   num2: 4,
-//   operation: 'add',
-// });
+const printResults = function(input, operator, result) {
+  console.log(`${input.num1} ${operator} ${input.num2} = ${result}`);
+};
+
+const addNums = function(input) {
+  const result = input.num1 * 1 + input.num2 * 1;
+  printResults(input, '+', result);
+};
+
+const subtractNums = function (input) {
+  const result = input.num1 * 1 - input.num2 * 1;
+  printResults(input, '-', result);
+};
+
+const multiplyNums = function (input) {
+  const result = input.num1 * input.num2
+  printResults(input, '*', result);
+};
 
+const divideNums = function(input) {
+  if(!(input.num2 * 1)) {
+    console.log('You cannot divide by zero');
+  } else {
+    const result = input.num1 / input.num2
+  printResults(input, '/', result);
+  }
+};
+
+const validNums = function(input) {
+  if (isNaN(input.num1) || isNaN(input.num2)) {
+    if (isNaN(input.num1)) {
+      console.log(`${input.num1} is not a number`);
+    }
+    if (isNaN(input.num2)) {
+      console.log(`${input.num2} is not a number`);
+    }
+  } else {
+    return true
+  }
+};
+
+const calculate = function(input) {
+  if (input.operation === 'add' || input.operation === '+') {
+    return addNums(input);
+  } else if (input.operation === 'subtract' || input.operation === '-') {
+    return subtractNums(input);
+  } else if (input.operation === 'multiply' || input.operation === '*') {
+    return multiplyNums(input);
+  } else if (input.operation === 'divide' || input.operation === '/') {
+    return divideNums(input);
+  } else {
+    console.log('unsupported operator');
```

Comment:
This works correctly, but generally when we write multiple `if` and `else if`s we want to use something like a switch/case statement! check them out here:

https://www.w3schools.com/js/js_switch.asp